File: Algorithms/121. Best Time to Buy and Sell Stock/Solution.go
package problem0121

func maxProfit(prices []int) int {
	length := len(prices)
	if length <= 1 {
		return 0
	}
	var min int
	var res int
	min = prices[0]
	for i := 1; i < length; i++ {
		temp := prices[i] - min
		if temp > res {
			res = temp
		}
		if prices[i] < min {
			min = prices[i]
		}
	}

	return res
}

File: Algorithms/121. Best Time to Buy and Sell Stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if prices == []:
            return 0
        currMin = prices[0]
        maxProfit = 0
        for i in range(len(prices)):
            currMin = min(currMin, prices[i])
            maxProfit = max(maxProfit, prices[i] - currMin)
        return maxProfit